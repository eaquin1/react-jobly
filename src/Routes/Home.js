import React, {useContext} from "react"
import img from "../Static/jobly.png"
import {Link} from "react-router-dom"
import UserContext from "../UserContext"
function Home() {
    const {currentUser} = useContext(UserContext)
    return (
        <div>
        <h1>Jobly</h1>
        <img src={img} alt="People working" style={{width: "900px"}}></img>
        {currentUser ? <h2>Welcome Back</h2> : <Link to="Login" style={{display: "block", fontSize: "3em"}}>Login</Link>}
        
        </div>
    )
    }

export default Home