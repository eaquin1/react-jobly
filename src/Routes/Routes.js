import React from "react";
import { Route, Switch, Redirect } from "react-router-dom"
import Companies from "../Components/Companies/Companies"
import Company from "../Components/Companies/Company"
import Jobs from "../Components/Jobs/Jobs"
import Login from "../Components/Auth/Login"
import Profile from "../Components/Profile/Profile"
import Home from "./Home"
import PrivateRoute from "./PrivateRoute"

function Routes({setToken}) {
    return (
        <Switch>     
            <PrivateRoute exact path="/companies"><Companies /></PrivateRoute>
            <PrivateRoute exact path="/companies/:handle"><Company /></PrivateRoute>
            <PrivateRoute exact path="/jobs"><Jobs /></PrivateRoute>
            <Route exact path="/login"><Login setToken={setToken} /></Route>
            <PrivateRoute exact path="/profile"><Profile /></PrivateRoute>
            <Route exact path="/"><Home /></Route>
            <Redirect to="/" />
        </Switch>
    )
}

export default Routes