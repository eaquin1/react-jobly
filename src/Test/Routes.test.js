import React from 'react';
import { render } from '@testing-library/react';
import Routes from '../Routes/Routes';
import { MemoryRouter } from 'react-router';
import { UserProvider } from '../testUtils';

describe('Testing the Routes', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider>
          <Routes />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Routes />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});