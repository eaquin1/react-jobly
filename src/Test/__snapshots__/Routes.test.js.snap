// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Routes matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Jobly
    </h1>
    <img
      alt="People working"
      src="jobly.png"
      style="width: 900px;"
    />
    <h2>
      Welcome Back
    </h2>
  </div>
</DocumentFragment>
`;
